eZ/Publish/Core/Repository/TrashService.php

```diff
@@ -63,15 +69,16 @@ public function __construct(
         RepositoryInterface $repository,
         Handler $handler,
         Helper\NameSchemaService $nameSchemaService,
-        array $settings = array()
+        array $settings = array(),
+        PermissionCriterionResolver $permissionCriterionResolver
```

Comment:
There is no BC promise for constructors so please move `$permissionCriterionResolver` before `$settings`